src/components/instructor/AssignmentAdd.js

```diff
@@ -1,19 +1,92 @@
 import React, { useState } from 'react';
+import { Button, Dialog, DialogActions, DialogContent, DialogTitle, TextField } from '@mui/material';
 
-// complete the code.  
+// complete the code.
 // instructor adds an assignment to a section
 // use mui Dialog with assignment fields Title and DueDate
 // issue a POST using URL /assignments to add the assignment
 
-const AssignmentAdd = (props)  => {
+const AssignmentAdd = (props) => {
+    const [open, setOpen] = useState(false);
+    const [title, setTitle] = useState('');
+    const [dueDate, setDueDate] = useState('');
+    const [message, setMessage] = useState('');
 
-   
+    const handleClickOpen = () => {
+        setOpen(true);
+    };
+
+    const handleClose = () => {
+        setOpen(false);
+        setTitle('');
+        setDueDate('');
+        setMessage('');
+    };
+
+    const handleSave = async () => {
+        const assignment = { title, dueDate };
+        try {
+            const response = await fetch(`${SERVER_URL}/assignments`, {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+                body: JSON.stringify(assignment),
+            });
+            if (response.ok) {
+                setMessage('Assignment added successfully');
+                props.save(assignment); // Assuming props.save will update the parent component's state
```

Comment:
Again, shouldn't be assuming things.

Reply:
I believe that the handleSave function should be used instead because you already defined it in assignmentUpdate.js. Props in this case, would be an assignment object and with the implementation of the other files, you should be able to see what functions those objects can call.